### PR TITLE
APPS-3336: Course benefits/UI container summary

### PR DESCRIPTION
**Pull request by rostikjoystick**
**YouTrack task**: [#APPS-3336](https://vyahhi.myjetbrains.com/youtrack/issue/APPS-3336)

**Description List**:
* Add course benefits container and feature
* Add course summary widget

### CODE REVIEW

File: app/src/main/res/layout/view_course_benefit_summary.xml [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginLeft="12dp"
+        android:layout_marginTop="8dp"
+        android:layout_marginRight="12dp"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintTop_toTopOf="parent">
+
+        <com.google.android.material.card.MaterialCardView
```
Comment (eadm) [inline app/src/main/res/layout/view_course_benefit_summary.xml:66]: Will it work on Android 5?

File: app/src/main/res/layout/view_course_benefit_summary.xml [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginLeft="4dp"
+            android:layout_marginTop="4dp"
+            android:layout_marginRight="4dp"
+            android:layout_marginBottom="4dp"
+            android:visibility="visible"
+            app:cardCornerRadius="8dp"
+            app:cardElevation="4dp"
+            app:layout_constraintLeft_toLeftOf="parent"
```
Comment (eadm) [inline app/src/main/res/layout/view_course_benefit_summary.xml:77]: ConstraintLayout attributes inside `RelativeLayout`

File: app/src/main/res/layout/view_course_benefit_summary.xml [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+                android:textAppearance="@style/TextAppearance.MaterialComponents.Caption"
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintLeft_toRightOf="@id/courseBenefitSummaryEarningsCurrentEmptyIcon"
+                app:layout_constraintTop_toTopOf="parent"
+                tools:text="Здесь будет ваш доход" />
+
+        </androidx.constraintlayout.widget.ConstraintLayout>
+    </com.google.android.material.card.MaterialCardView>
+
+    <RelativeLayout
```
Comment (eadm) [inline app/src/main/res/layout/view_course_benefit_summary.xml:55]: Can we use LinearLayout?

File: app/src/main/java/org/stepik/android/view/course_benefits/ui/delegate/CourseBenefitSummaryViewDelegate.kt [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+            ).capitalize(Locale.ROOT)
+
+            val totalDate = DateTimeHelper.getPrintableDate(
+                state.courseBenefitSummary.beginPaymentDate,
+                DateTimeHelper.DISPLAY_MONTH_YEAR_GENITIVE_PATTERN,
+                TimeZone.getDefault()
+            ).capitalize(Locale.ROOT)
+
+            courseBenefitCurrentEarningsTitle.text = context.getString(R.string.course_benefits_earning_current_month, currentMonthDate)
+            courseBenefitCurrentEarningsValue.text = "129000 ₽"
```
Comment (eadm) [inline app/src/main/java/org/stepik/android/view/course_benefits/ui/delegate/CourseBenefitSummaryViewDelegate.kt:110]: test data?

File: app/src/main/res/layout/view_course_benefit_summary.xml (hunk 1 above)
Comment (rostikjoystick) [reply app/src/main/res/layout/view_course_benefit_summary.xml:66]: Checked on API 21 emulator, ripple and expand/collapse work fine

**Review by eadm — APPROVED** [review]
- remove toolbar elevation & add separator
- up button left margin should be 16dp

**Pull request merged**